src/migrateDashboard.ts

```diff
@@ -180,6 +194,30 @@ export function migrateDashboard(
     queryContext: _migrateContextValues(body.contextValues),
   };
 
+  // If the dashboard contains a disconnected widget, then move the dashboard filters down to every page.
+  // If a given page contains a disconnected widget, then move the page filters down to every connected widget.
+  if (!_isEmpty(keysOfDisconnectedWidgets)) {
+    const dashboardFilters = dashboard.filters;
+    delete dashboard.filters;
+
+    Object.entries(pages).forEach(([pageKey, page]) => {
+      page.filters = [...(dashboardFilters ?? []), ...(page.filters ?? [])];
+      if (!_isEmpty(keysOfDisconnectedWidgets[pageKey])) {
+        const pageFilters = page.filters;
+        delete page.filters;
+
+        Object.entries(page.content).forEach(([leafKey, widget]) => {
+          if (!keysOfDisconnectedWidgets[pageKey].includes(leafKey)) {
```

Comment:
Nitpicky, but this could be made more performant if `keysOfDisconnectedWidgets` contained sets instead of arrays.